File: core/src/ua/syt0r/Assets.java
package ua.syt0r;

import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.graphics.g2d.TextureAtlas;
import com.badlogic.gdx.graphics.g2d.TextureRegion;

public class Assets {

    private static final AssetManager assetManager = new AssetManager();

    public static TextureAtlas loadingAtlas;
    public static TextureAtlas uiAtlas;
    public static TextureAtlas gameAtlas;

    public static void loadLoadingScreenAssets(){

        assetManager.load("loading.atlas",TextureAtlas.class);
        assetManager.finishLoading();
        loadingAtlas = Assets.get("loading.atlas", TextureAtlas.class);

    }

    public static void loadMenuUIAssets(){

        assetManager.load("ui.atlas",TextureAtlas.class);
        assetManager.finishLoading();
        uiAtlas = Assets.get("ui.atlas", TextureAtlas.class);

    }

    public static void unload(String resourceName){

        assetManager.unload(resourceName);

    }

    public static void load(String fileName, Class<?> classType){
        assetManager.load(fileName,classType);
    }

    public static <T> T get(String resourceName, Class<T> classType){
        return assetManager.get(resourceName,classType);
    }

    public static boolean update(){
        return assetManager.update();
    }

    public static void dispose(){
        assetManager.dispose();
    }



}

File: core/src/ua/syt0r/Assets.kt
package ua.syt0r

import com.badlogic.gdx.assets.AssetManager
import com.badlogic.gdx.graphics.g2d.TextureAtlas

object Assets {

    private val assetManager = AssetManager()

    lateinit var loadingAtlas: TextureAtlas
    lateinit var uiAtlas: TextureAtlas
    lateinit var gameAtlas: TextureAtlas

    @JvmStatic
    fun loadLoadingScreenAssets() {
        assetManager.load("loading.atlas", TextureAtlas::class.java)
        assetManager.finishLoading()
        loadingAtlas = get("loading.atlas", TextureAtlas::class.java)
    }

    @JvmStatic
    fun loadMenuUIAssets() {
        assetManager.load("ui.atlas", TextureAtlas::class.java)
        assetManager.finishLoading()
        uiAtlas = get("ui.atlas", TextureAtlas::class.java)
    }

    @JvmStatic
    fun loadGameAssets() {
        assetManager.load("game.atlas", TextureAtlas::class.java)
        assetManager.finishLoading()
        gameAtlas = get("game.atlas", TextureAtlas::class.java)
    }

    @JvmStatic
    fun unload(resourceName: String?) {
        assetManager.unload(resourceName)
    }

    @JvmStatic
    fun load(fileName: String?, classType: Class<*>?) {
        assetManager.load(fileName, classType)
    }

    @JvmStatic
    operator fun <T> get(resourceName: String?, classType: Class<T>?): T {
        return assetManager.get(resourceName, classType)
    }

    @JvmStatic
    fun update(): Boolean {
        return assetManager.update()
    }

    @JvmStatic
    fun dispose() {
        assetManager.dispose()
    }
}